app/controllers/metrics_controller.rb

```diff
@@ -41,4 +41,42 @@ def get_watchlist_instances
 		end
 	end
 
+	action_auth_level :update_current_metrics, :instructor
+	def update_current_metrics
+		# This API endpoint aims to update current/latest risk conditions for a particular course
+		# On success, a JSON list of condition objects that are freshly created out of the request parameters will be returned
+		# The request parameters must take on the following format:
+		# params[type_a] is nil if instructor does not want to include a risk condition of type_a during update
+		# params[type_a] is an object satisfying the parameter requirement for type_a condition
+		# e.g. params["grace_day_usage"] should have form { "grace_day_threshold" => 3, "date" => "2020-03-05" }
+		# Currently, four types of risk conditions are supported:
+		# "grace_day_usage" with parameters "grace_day_threshold", "date"
+		# "grade_drop" with parameters "percentage_drop", "consecutive_counts"
+		# "no_submissions" with parameter "no_submissions_threshold"
+		# "low_grades" with parameters "grade_threshold", "count_threshold"
+		# On error, a flash error message will be rendered and nil gets returned
+		begin
+			course_name = params[:course_name]
+			params_filtered = new_metrics_params
+			if params_filtered.nil?
+				params_filtered = {}
+			else
+				params_filtered = params_filtered.to_h
+			end
+			conditions = RiskCondition.update_current_for_course(course_name, params_filtered)
+			render json: conditions
+		rescue => error
```

Comment:
Do not flash the error message, because it is not rendering the page, so the flashing of the error cannot be shown, instead return an appropriate status instead.

```
rescue => error
	render :text => 'Not Found', :status => '404'
	return
end
```

refer to here for the status codes
https://gist.github.com/mlanett/a31c340b132ddefa9cca

Reply:
See latest commit.